File: backend/internal/model/notification.go
package model

import "time"

type Notification struct {
	ID           uint64     `db:"id"            json:"id"`
	UserID       uint64     `db:"user_id"       json:"user_id"`
	Type         string     `db:"type"          json:"type"`
	Title        string     `db:"title"         json:"title"`
	Body         string     `db:"body"          json:"body"`
	ResourceType *string    `db:"resource_type" json:"resource_type,omitempty"`
	ResourceID   *uint64    `db:"resource_id"   json:"resource_id,omitempty"`
	IsRead       bool       `db:"is_read"       json:"is_read"`
	CreatedAt    time.Time  `db:"created_at"    json:"created_at"`
}

File: frontend/src/shared/types/notification.ts
export type NotificationItem = {
  id: number
  user_id: number
  type: string
  title: string
  body: string
  resource_type?: string | null
  resource_id?: number | null
  resource_ref?: string | null
  is_read: boolean
  created_at: string
}

export type NotificationListResponse = {
  notifications: NotificationItem[]
  total: number
  unread: number
  limit: number
  offset: number
}

export type NotificationSummary = {
  pending: number
  review_required: number
  execution_required: number
}
